versioned_docs/version-4/generate-test-paths.mdx

```diff
@@ -0,0 +1,31 @@
+---
+title: Generate test paths
+---
+
+You can generate test path code for your state machines from the <ClipboardCheck size={18}/> **Tests** panel, which you can use to implement tests in your code. Both setup code and test code for each path is generated.
+
+This feature is currently in beta and [we’d love to hear your feedback](https://discord.stately.ai).
+
+:::studio
+
+**Generate test paths** is a paid feature of Stately Studio. You can try Stately Studio’s paid plans with a free trial. [Check out the features on our Pro plan](studio-pro-plan.mdx), [Team plan](studio-team-plan.mdx), [Enterprise plan](studio-enterprise-plan.mdx) or [upgrade your existing plan](https://stately.ai/registry/billing).
+
+:::
+
+## Path generation strategy
+
+You can choose between two path generation strategies:
+
+- **Shortest path**: Generate paths that cover the shortest possible paths through your state machine and minimizes the number of actions required to test functionality.
+- **Simplest path**: Generate paths that cover the simplest paths, often known as “happy paths”, through your state machine and helps verify that basic functionality is working as expected.
```

Comment:
```suggestion
- **Simplest path**: Generate paths that cover the simple paths (paths that contain no cycles) through your state machine and helps verify that basic functionality is working as expected.
```

Reply:
@davidkpiano Will it be clear to users what "cycles" are or should we add or replace it with "repeated states" or "states are visited only once"?